### PR TITLE
PRO-6069: Tests optimisation

### PR DESCRIPTION
**Pull request by LucaDelBuonoHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-6069

### Change description ###
Optimisation of component and unit tests

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: app/resources/en/translation/addressLookup.json [hunk 1]
```diff
@@ -6,11 +6,11 @@
         "message": "Please enter a postcode"
```
Comment (douglasrice) [inline app/resources/en/translation/addressLookup.json]: This is not an optimisation and should be raised in a separate ticket

File: app/resources/en/translation/payment/status.json [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-    "paragraph": "We&rsquo;ve received your payment and your application.",
+    "paragraph1": "We&rsquo;ve received your payment and your application.",
     "paragraph2": "We&rsquo;ve received your application.",
     "paragraph3": "You don&rsquo;t need to pay as the net value of the estate is less than &pound;5,000.",
     "callout": "Before your application can be processed, you need to send your documents by post.",
     "errors": {
         "update": {
             "failure": {
-                "summary": "We could not submit your application. Your data has been saved, please try again later."
+                "summary": "We could not submit your application. Your data has been saved, please try again later.",
```
Comment (douglasrice) [inline app/resources/en/translation/payment/status.json]: these are not optimisations and should be raised as a separate ticket

Comment (LucaDelBuonoHMCTS) [reply app/resources/en/translation/payment/status.json]: This one actually is a bug fix. The change on the above line is the comma at the end as I added the missing `message` property below.

File: app/resources/en/translation/addressLookup.json (hunk 1 above)
Comment (LucaDelBuonoHMCTS) [reply app/resources/en/translation/addressLookup.json]: Ok I'll revert this

File: app/steps/ui/deceased/address/template.html [hunk 3]
```diff
@@ -23,7 +23,7 @@
             <summary class="govuk-details__summary">
                 <span class="govuk-details__summary-text">{{ content.enterManually }}</span>
             </summary>
-            <div class="govuk-details__text">
+            <div class="govuk-form-group govuk-details__text">
```
Comment (douglasrice) [inline app/steps/ui/deceased/address/template.html]: Remove this code and new ticket to be raised for GDS changes

File: app/steps/ui/executors/address/template.html [hunk 4]
```diff
@@ -24,7 +24,7 @@
             <summary class="govuk-details__summary">
                 <span class="govuk-details__summary-text">{{ content.enterManually }}</span>
             </summary>
-            <div class="govuk-details__text">
+            <div class="govuk-form-group govuk-details__text">
```
Comment (douglasrice) [inline app/steps/ui/executors/address/template.html]: Remove this code and new ticket to be raised for GDS changes

File: app/steps/ui/payment/status/template.html [hunk 5]
```diff
@@ -8,7 +8,7 @@ <h1 class="govuk-heading-l">{{ content.question }}</h1>
 
     {% if errors == undefined %}
         {% if fields.paymentDue.value == "true" %}
-            <p class="govuk-body">{{ content.paragraph | safe }}</p>
+            <p class="govuk-body">{{ content.paragraph1 | safe }}</p>
```
Comment (douglasrice) [inline app/steps/ui/payment/status/template.html]: Not an optimisation

Comment (LucaDelBuonoHMCTS) [reply app/steps/ui/payment/status/template.html]: It was also changed it in the translation file, won't affect anything, leave this.

File: app/views/widgets/addresslookup.html [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
         <input type="text" class="govuk-input govuk-!-width-one-quarter {{ "govuk-input--error" if fields.postcode.error }}" id="postcode" name="postcode" value="{{ fields.postcode.value }}">
         <input type="hidden" id="referrer" name="referrer" value="{{ step }}">
-        <input type="submit" data-prevent-double-click="true" class="govuk-button govuk-button--secondary find-address-button" formaction="/find-address" name="findaddress" value="{{ content.findAddress }}">
+        <input type="submit" data-prevent-double-click="true" class="govuk-button govuk-button--secondary find-address-button" formaction="/find-address" value="{{ content.findAddress }}">
```
Comment (douglasrice) [inline app/views/widgets/addresslookup.html]: new ticket please

Comment (LucaDelBuonoHMCTS) [reply app/views/widgets/addresslookup.html]: That HTML property is unused, just removing it is an optimisation.

File: app/views/widgets/fielderror.html [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {% macro fieldError(field) %}
     {% if field.error %}
         <span class="govuk-error-message">
-            {{ field.errorMessage.text | safe }}
+            {{ field.errorMessage.message | safe }}
```
Comment (douglasrice) [inline app/views/widgets/fielderror.html]: new ticket please

File: test/component/screeners/testApplicantExecutor.js [hunk 8]
```diff
@@ -34,10 +34,10 @@ describe('applicant-executor', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('ApplicantExecutor', null, cookies);
+        testCommonContent.runTest('ApplicantExecutor', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testApplicantExecutor.js]: too many parameters in call

File: test/component/screeners/testDeathCertificate.js [hunk 9]
```diff
@@ -21,14 +21,16 @@ describe('death-certificate', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('DeathCertificate');
+        testCommonContent.runTest('DeathCertificate', null, null, [], true);
```
Comment (douglasrice) [inline test/component/screeners/testDeathCertificate.js]: too many parameters

File: test/component/screeners/testDeceasedDomicile.js [hunk 10]
```diff
@@ -30,10 +30,10 @@ describe('deceased-domicile', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('DeceasedDomicile', null, cookies);
+        testCommonContent.runTest('DeceasedDomicile', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testDeceasedDomicile.js]: too many parameters in call

File: test/component/screeners/testIhtCompleted.js [hunk 11]
```diff
@@ -31,10 +31,10 @@ describe('iht-completed', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('IhtCompleted', null, cookies);
+        testCommonContent.runTest('IhtCompleted', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testIhtCompleted.js]: too many parameters for runTest

File: test/component/screeners/testMentalCapacity.js [hunk 12]
```diff
@@ -35,10 +35,14 @@ describe('mental-capacity', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('MentalCapacity', null, cookies);
+        testCommonContent.runTest('MentalCapacity', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testMentalCapacity.js]: too many parameters for runTest

File: test/component/screeners/testOtherApplicants.js [hunk 13]
```diff
@@ -36,10 +36,10 @@ describe('other-applicants', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('OtherApplicants', null, cookies);
+        testCommonContent.runTest('OtherApplicants', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testOtherApplicants.js]: too many parameters for runTest

File: test/component/screeners/testRelatedToDeceased.js [hunk 14]
```diff
@@ -34,10 +34,10 @@ describe('related-to-deceased', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('RelatedToDeceased', null, cookies);
+        testCommonContent.runTest('RelatedToDeceased', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testRelatedToDeceased.js]: too many parameters for runTest

File: test/component/screeners/testWillLeft.js [hunk 15]
```diff
@@ -32,10 +32,10 @@ describe('will-left', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('WillLeft', null, cookies);
+        testCommonContent.runTest('WillLeft', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testWillLeft.js]: too many parameters for runTest

File: test/component/screeners/testWillOriginal.js [hunk 16]
```diff
@@ -33,10 +33,10 @@ describe('will-original', () => {
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('WillOriginal', null, cookies);
+        testCommonContent.runTest('WillOriginal', null, null, cookies, true);
```
Comment (douglasrice) [inline test/component/screeners/testWillOriginal.js]: too many parameters for runTest

File: test/component/summary/testDeceasedSection.js [hunk 17]
```diff
@@ -70,11 +77,11 @@ describe('summary-deceased-section', () => {
                         married: deceasedContent.married.question.replace('{deceasedName}', deceasedName),
                         dob: deceasedContent.dob.question,
                         dod: deceasedContent.dod.question,
-                        address: deceasedContent.address.question,
+                        address: deceasedData.deceased.address.formattedAddress,
```
Comment (douglasrice) [inline test/component/summary/testDeceasedSection.js]: This section of the test needs further investigation as it validates the questions being asked bu the answers seem to be missing.....

File: test/component/summary/testExecutorsSection.js [hunk 18]
```diff
@@ -99,30 +98,31 @@ describe('summary-executor-section', () => {
...
                 .end((err) => {
                     if (err) {
                         throw err;
                     }
 
+                    delete require.cache[require.resolve('test/data/summary-executors')];
                     const playbackData = {
                         firstName: applicantContent.name.firstName,
                         lastName: applicantContent.name.lastName,
                         phoneNumber: applicantContent.phone.phoneNumber,
```
Comment (douglasrice) [inline test/component/summary/testExecutorsSection.js]: Missing actual phone number and also email for apply additional executors

File: test/component/summary/testExecutorsSection.js [hunk 19]
```diff
@@ -99,30 +98,31 @@ describe('summary-executor-section', () => {
...
+                        applicantAddress: applicantContent.address.question,
+                        address: executorsData.applicant.address.formattedAddress,
+                        exec2fullName: executorsData.executors.list[1].fullName,
+                        exec2IsApplying: executorsData.executors.list[1].isApplying ? 'Yes' : 'No',
+                        exec2HasAlias: executorsData.executors.list[1].hasOtherName ? 'Yes': 'No',
+                        exec2Alias: executorsData.executors.list[1].currentName,
+                        exec2AliasReason: executorsData.executors.list[1].currentNameReason,
+                        exec3fullName: executorsData.executors.list[2].fullName,
+                        exec3IsApplying: executorsData.executors.list[2].isApplying ? 'Yes' : 'No',
+                        exec3NotApplyingReason: executorsData.executors.list[2].notApplyingReason
```
Comment (douglasrice) [inline test/component/summary/testExecutorsSection.js]: need to add additional data to this test for a deceased executor ie "isDead": true,

File: test/component/testDocumentsPage.js [hunk 20]
```diff
@@ -439,48 +415,36 @@ describe('documents', () => {
...
-                            'checklist-item3-will-uploaded',
-                            'checklist-item4-iht205',
-                            'checklist-item5-renunciated',
-                            'checklist-item6-deed-poll',
-                            'checkboxLabel-codicils',
-                            'checkboxLabel'
-                        ];
-
-                        testWrapper.testContent(done, excludeKeys, contentData);
+                        testWrapper.testDataPlayback(done);
```
Comment (douglasrice) [inline test/component/testDocumentsPage.js]: This test needs changing as should be using assertContentNotPresent

File: test/unit/testAddressLookup.js [hunk 21]
```diff
@@ -43,7 +42,7 @@ describe('AddressLookup', () => {
             referrer: 'ApplicantAddress',
             postcode: 'SW1H 9AJ'
         };
-        errorsToTest = {};
+        errorsToTest = [];
```
Comment (douglasrice) [inline test/unit/testAddressLookup.js]: declared an array but then assigned an object later on - inconsistent

File: test/unit/testAddAlias.js [hunk 22]
```diff
@@ -4,7 +4,7 @@ const initSteps = require('app/core/initSteps');
 const assert = require('chai').assert;
 
 describe('AddAlias', () => {
-    const steps = initSteps([`${__dirname}'/../../app/steps/action/`, `${__dirname}/../../app/steps/ui`]);
+    const steps = initSteps([`${__dirname}/../../app/steps/action/`, `${__dirname}/../../app/steps/ui/`]);
```
Comment (douglasrice) [inline test/unit/testAddAlias.js]: Additional '/' added to app/steps/ui/   - I think this isn't required - please confirm

File: test/unit/testCoApplicant.js [hunk 23]
```diff
@@ -1,10 +1,9 @@
-const initSteps = require('app/core/initSteps'),
-    assert = require('chai').assert,
+const initSteps = require('app/core/initSteps');
+const assert = require('chai').assert,
     {isNil} = require('lodash');
```
Comment (douglasrice) [inline test/unit/testCoApplicant.js]: assert variable has been changed to be individually declared as const (was previously declared in tandem with initSteps.  Thus to keep this standardised we should also declare {isNil} individually as a const

File: test/component/coapplicant/testCoApplicantDeclaration.js [hunk 24]
```diff
@@ -4,66 +4,69 @@ const TestWrapper = require('test/util/TestWrapper');
...
 
     afterEach(() => {
+        delete require.cache[require.resolve('test/data/complete-form-undeclared')];
         nock.cleanAll();
         testWrapper.destroy();
     });
 
     describe('Verify Content, Errors and Redirection', () => {
-        testHelpBlockContent.runTest('CoApplicantDeclaration', invitesNock);
+        testCommonContent.runTest('CoApplicantDeclaration', null, null, [], true);
```
Comment (douglasrice) [inline test/component/coapplicant/testCoApplicantDeclaration.js]: too many parameters

**Pull request merged**